index.js

```diff
@@ -191,48 +191,38 @@ NYC.prototype.addAllFiles = function () {
 }
 
 NYC.prototype.instrumentAllFiles = function (input, output, cb) {
-  var _this = this
-  var inputDir = '.' + path.sep
-  var visitor = function (filename) {
-    var ext
-    var transform
-    var inFile = path.resolve(inputDir, filename)
-    var code = fs.readFileSync(inFile, 'utf-8')
-
-    for (ext in _this.transforms) {
-      if (filename.toLowerCase().substr(-ext.length) === ext) {
-        transform = _this.transforms[ext]
-        break
-      }
-    }
+  const visitor = (filename, singleFile = false) => {
+    const inFile = path.resolve(input, filename)
+    const inCode = fs.readFileSync(inFile, 'utf-8')
 
-    if (transform) {
-      code = transform(code, { filename: inFile, relFile: inFile })
-    }
+    const extname = path.extname(filename).toLowerCase()
+    const transform = this.transforms[extname] || (code => code)
+    const outCode = transform(inCode, { filename: inFile })
 
-    if (!output) {
-      console.log(code)
-    } else {
-      var outFile = path.resolve(output, filename)
+    if (output) {
+      const subPath = singleFile || path.relative(input, path.resolve(input, filename))
+      const outFile = path.resolve(output, subPath)
       mkdirp.sync(path.dirname(outFile))
-      fs.writeFileSync(outFile, code, 'utf-8')
+      fs.writeFileSync(outFile, outCode, 'utf-8')
+    } else {
+      console.log(outCode)
     }
   }
 
   this._loadAdditionalModules()
 
   try {
-    var stats = fs.lstatSync(input)
+    const stats = fs.lstatSync(input)
     if (stats.isDirectory()) {
-      inputDir = input
       this.walkAllFiles(input, visitor)
     } else {
-      visitor(input)
+      const inputPath = path.resolve('.', input)
```

Comment:
The `'.'` argument is redundant / always implied by `path.resolve`.  Though if this will be replaced by the `--cwd` option then don't worry about change it.

Reply:
I'll change it for correctness